goal_src/levels/sunken/sunken-obs.gc

```diff
@@ -5,16 +5,845 @@
+        )
+       (set!
+        (-> a0-39 param 0)
+        (the
+         float
+         (+
+          (->
+           (the-as art-joint-anim (-> self draw art-group data 17))
+           data
+           0
+           length
+           )
+          -1
+          )
+         )
+        )
+       (set! (-> a0-39 param 1) 0.67)
+       (set! (-> a0-39 frame-num) 0.0)
+       (joint-control-channel-group!
+        a0-39
+        (the-as art-joint-anim (-> self draw art-group data 17))
+        num-func-seek!
+        )
+       )
+      (until (ja-done? 0)
+       (suspend)
+       (let ((a0-40 (-> self skel root-channel 0)))
+        (set!
+         (-> a0-40 param 0)
+         (the float (+ (-> a0-40 frame-group data 0 length) -1))
+         )
+        (set! (-> a0-40 param 1) 0.67)
+        (joint-control-channel-group-eval!
+         a0-40
+         (the-as art-joint-anim #f)
+         num-func-seek!
+         )
+        )
+       )
+      (pov-camera-play-and-reposition
+       (the-as joint-anim-compressed (-> self draw art-group data 18))
+       (new 'static 'vector :x 2449408.0 :y -1822720.0 :z -7299072.0)
+       0.67
+       )
+      )
+     ((= v1-0 3)
+      (ambient-hint-spawn "gamcam34" (the-as vector #f) *entity-pool* 'camera)
+      (pov-camera-play-and-reposition
+       (the-as joint-anim-compressed (-> self draw art-group data 12))
+       (new 'static 'vector :x 2707456.0 :y -630784.0 :z -6635520.0)
+       1.0
+       )
+      )
+     ((= v1-0 4)
+      (ambient-hint-spawn "gamcam35" (the-as vector #f) *entity-pool* 'camera)
+      (pov-camera-play-and-reposition
+       (the-as joint-anim-compressed (-> self draw art-group data 13))
+       (new 'static 'vector :x 2834432.0 :y -634880.0 :z -6811648.0)
+       1.3
+       )
+      (set! (-> self state-time) (-> *display* base-frame-counter))
+      (until (>= (- (-> *display* base-frame-counter) (-> self state-time)) 60)
+       (suspend)
+       )
+      )
+     ((= v1-0 5)
+      (ambient-hint-spawn "gamcam37" (the-as vector #f) *entity-pool* 'camera)
+      (pov-camera-play-and-reposition
+       (the-as joint-anim-compressed (-> self draw art-group data 14))
+       (new 'static 'vector :x 2695168.0 :y -544768.0 :z -7077888.0)
+       1.0
+       )
+      )
+     ((= v1-0 6)
+      (ambient-hint-spawn "gamcam36" (the-as vector #f) *entity-pool* 'camera)
+      (pov-camera-play-and-reposition
+       (the-as joint-anim-compressed (-> self draw art-group data 15))
+       (new 'static 'vector :x 2482176.0 :y -516096.0 :z -6922240.0)
+       1.0
+       )
+      )
+     (else
+      (suspend)
+      0
+      )
+     )
+    )
+   (go-virtual pov-camera-done-playing)
+   (none)
+   )
+  )
+
+(deftype seaweed (process-drawable)
+  ((anim-speed float  :offset-assert 176)
+   )
+  :heap-base #x50
+  :method-count-assert 20
+  :size-assert         #xb4
+  :flag-assert         #x14005000b4
+  )
+
+;; TODO - manually added
+(define-perm *seaweed* seaweed #f)
```

Comment:
we might want to keep this as a `(define-extern *seaweed* seaweed)` unless there's a specific reason to do this.  There might be code that checks the value of `*seaweed*` before anything sets it, and the default value should be 0, not `#f`.